wremnants/CardTool.py

```diff
@@ -492,7 +585,12 @@ def writeOutput(self, args=None):
         self.writeForProcesses(self.nominalName, processes=self.datagroups.groups.keys(), label=self.nominalName)
         self.loadNominalCard()
         if self.pseudoData and not self.xnorm:
-            self.addPseudodata([x for x in self.datagroups.groups.keys() if x != "Data"])
+            self.addPseudodata([x for x in self.procDict.keys() if x != "Data"],
+                               [x for x in self.procDict.keys() if x != "Data" and not self.pseudoDataProcsRegexp.match(x)])
+        ## NOTE: when fixing conflicts I noticed the following had self.datagroups.groups.keys() instead of self.procDict.keys()
+        ##       should be the same but I am not sure if there are unexpected features with one or the other case
```

Comment:
It seems to me that there is some inconsistency, probably self.procDict was dropped because it didn't make sense to track it separately from self.datagroups

Reply:
Ok, I'll use the datagroups list. It should be the same as long as it takes into account possible filters passed from the beginning when creating the datagroups (which I believe it does)